components/server/ee/src/workspace/gitpod-server-impl.ts

```diff
@@ -2190,6 +2194,22 @@ export class GitpodServerEEImpl extends GitpodServerImpl {
         };
     }
 
+    async getBillingModeForUser(ctx: TraceContextWithSpan): Promise<BillingMode> {
+        traceAPIParams(ctx, {});
+
+        const user = this.checkUser("getBillingModeForUser");
+        return this.billingModes.getBillingModeForUser(user, new Date());
```

Comment:
Passing `new Date()` here feels a little odd, in that I wouldn't expect a user or team's billing mode to depend on the current time.

Reply:
It does, and other methods take it as well.

Thing is, I tried to remove it here by making it optional (defaulting to `new Date()`), but that does **not** result in compile errors when you implement it wrongly - thank you, Javascript. :face_with_spiral_eyes:

Reply:
I'm all-in for streamlinig the API, but considered it out-of-scope because of the extra-effort to avoid runtime issues.

Reply:
yeah, making it default to the current time was what I was getting at, but if that's more trouble than it's worth ✅